src/components/Editor/QuillEditor.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useRef, useState } from "react";
+import "../../css/quillEditor.css";
+import { useFirebase } from "react-redux-firebase";
+import { useDispatch, useSelector } from "react-redux";
+import { Prompt } from "react-router-dom";
+import { setCurrentStep } from "../../store/actions";
+import * as Y from "yjs";
+import { QuillBinding } from "y-quill";
+import Quill from "quill";
+import QuillCursors from "quill-cursors";
+import { FirestoreProvider, getColor } from "@gmcfall/yjs-firestore-provider";
+
+import { onlineFirebaseApp } from "../../config";
+
+Quill.register("modules/cursors", QuillCursors);
+
+const QuillEditor = ({ id, data, tutorial_id }) => {
+  const [allSaved, setAllSaved] = useState(true);
+  const editorRef = useRef(null);
+  const containerRef = useRef(null);
+  let noteID = id || "test_note";
+  const firebase = useFirebase();
+  const dispatch = useDispatch();
+  const basePath = ["cl_codelabz", "organization", "codelabzorg", tutorial_id];
+  var provider, binding;
+  var ydoc;
+
+  const currentUserHandle = useSelector(
+    ({
+      firebase: {
+        profile: { handle }
+      }
+    }) => handle
+  );
+
+  useEffect(() => {
+    setAllSaved(true);
+  }, [id]);
+
+  useEffect(() => {
+    try {
+      if (!ydoc) {
+        ydoc = new Y.Doc();
+
+        // on updating text in editor this gets triggered
+        ydoc.on("update", () => {
+          const deltaText = ydoc.getText("quill").toDelta();
+          setCurrentStep(deltaText)(dispatch);
+        });
+        provider = new FirestoreProvider(onlineFirebaseApp, ydoc, basePath, {
+          disableAwareness: true
+        });
+      }
+      const ytext = ydoc.getText("quill");
+      const container = containerRef.current;
+
+      // Clear all extra divs except the editor
+      while (
+        container.firstChild &&
+        container.firstChild !== editorRef.current
+      ) {
+        container.removeChild(container.firstChild);
+      }
+
+      const editor = new Quill(editorRef.current, {
+        modules: {
+          cursors: true,
+          toolbar: [
+            [{ header: [1, 2, false] }],
+            ["bold", "italic", "underline"],
+            ["image", "code-block"]
+          ],
+          history: {
+            userOnly: true
+          }
+        },
+        placeholder: "Start collaborating...",
+        theme: "snow"
+      });
+
+      // provider.awareness.setLocalStateField("user", {
+      //   name: currentUserHandle,
+      //   color: getColor(currentUserHandle)
+      // });
+
```

Comment:
Not sure why we have left this code as commented?

Reply:
We are not using the concept of awarness for now but may use in future, so left it.

Reply:
Oh okay. Got it! 
Can we then add this same comment in the code?

Reply:
I'll soon be un commenting that in my next few prs.

Reply:
Sounds cool.

Reply:
Btw out of this context, is there any frontend engineering or any other issue ticket for me open? 
I want to contribute! 😄 Looking forward to it. 
I see gitter as quite in-active, can you please help me?